internal/servers/controller/handlers/targets/registry.go

```diff
@@ -56,45 +60,39 @@ func (r registry) maskManager(s subtypes.Subtype) (handlers.MaskManager, error)
 
 // newAttribute creates an Attribute for the given subtype. It delegates the
 // allocation of the Attribute to the registered attrFunc for the given
-// subtype. An error is returned if the provided subtype is not registered. It
-// supports the options of:
-//
-// - withTarget: The Attribute will be initialized based on the given
-// target.Target. This initialization is delegated to the attrFunc.
-//
-// - withStruct: The Attribute will be initialized based on the given
-// structpb.Struct. This is *not* delegated.
-//
-// These two options are mutually exclusive. If both are provided, and error is returned.
-// If no options are provided an empty Attribute will be returned.
-func (r registry) newAttribute(s subtypes.Subtype, opt ...option) (Attributes, error) {
+// subtype. An error is returned if the provided subtype is not registered
+func (r registry) newAttribute(s subtypes.Subtype, m interface{}) (Attributes, error) {
 	re, err := r.get(s)
 	if err != nil {
 		return nil, err
 	}
 
-	opts := getOpts(opt...)
-	if opts.withTarget != nil && opts.withStruct != nil {
-		return nil, fmt.Errorf("cannot use both withTarget and withStruct")
-	}
+	attr := re.attrFunc(m)
+
+	return attr, nil
+}
 
-	attr := re.attrFunc(opts.withTarget)
-	if opts.withStruct != nil {
-		if err := handlers.StructToProto(opts.withStruct, attr); err != nil {
-			return nil, fmt.Errorf("Provided attributes don't match expected format.")
-		}
+// setAttributes is used to set the Attrs field on a pb.Target. It delegates the
+// setting of the specific attribute type to the registered setAttirbuteFunc for
+// the given subtype. An error is returned if the provided subtype is not
+// registered.
+func (r registry) setAttributes(s subtypes.Subtype, in target.Target, out *pb.Target) error {
+	re, err := r.get(s)
+	if err != nil {
+		return err
 	}
 
-	return attr, nil
+	return re.setAttrFunc(in, out)
 }
 
 var subtypeRegistry = registry{}
 
 // Register registers a subtype for used by the service handler.
-func Register(s subtypes.Subtype, maskManager handlers.MaskManager, af attributeFunc) {
+func Register(s subtypes.Subtype, maskManager handlers.MaskManager, af attributeFunc, sf setAttributeFunc) {
```

Comment:
Do we plan on keeping this around after the merge to enterprise? I'm not sure I understand the full scope of this but should we not be able to rip all this out and just switch on subtype like we do in other parts of the code?